lib/dor/workflow/client/queues.rb

```diff
@@ -11,53 +11,55 @@ def initialize(requestor:)
 
         # Returns all the distinct laneIds for a given workflow step
         #
-        # @param [String] repo The repository the object resides in.  The service recoginzes "dor" and "sdr" at the moment
+        # @param [String] repo -- deprecated, ignored by workflow service
         # @param [String] workflow name
         # @param [String] process name
         # @return [Array<String>] all of the distinct laneIds.  Array will be empty if no lane ids were found
         def lane_ids(repo, workflow, process)
-          uri = "workflow_queue/lane_ids?step=#{repo}:#{workflow}:#{process}"
+          Deprecation.warn(self, 'the first argument to `#lane_ids` is unused and will go away in Dor::Workflow::Client 4.0.0. pass `nil` to silence.') unless repo.nil?
```

Comment:
Should this do what @aaron-collier has been doing (set `*args`) and warn if the count is 3?

Reply:
@jcoyne I'd be inclined to call this good enough for now, but I'm open to making more changes especially if you can help me grok the benefit of this.

Reply:
@jcoyne I'd be inclined to call this good enough for now, but I'm open to making more changes especially if you can help me grok the benefit of this.

Reply:
Won't this be problematic if the first argument is the workflow since they are positional?

Reply:
Oh I see, you're passing `nil` below.

Reply:
Doesn't is seem silly to have a method that you have to pass `nil` as the first method?

Reply:
If you remove this argument in 4.0, it will break any callers still passing 3 args. If you make the number of arguments optional, consumers can get ready for the change ahead of time.

Reply:
> If you remove this argument in 4.0, it will break any callers still passing 3 args. If you make the number of arguments optional, consumers can get ready for the change ahead of time.

I mean, what fun is releasing a new major version if you don't break old callers that haven't correctly pinned a dependency?

I'll make the change. Requiring clients to pass a nil is pretty silly.